Updated encoder config docs & standardization of existing shields (#1970)
zmkfirmware/zmk: docs/docs/development/new-shield.md

        label = "LEFT_ENCODER";
        a-gpios = <PIN_A (GPIO_ACTIVE_HIGH | GPIO_PULL_UP)>;
        b-gpios = <PIN_B (GPIO_ACTIVE_HIGH | GPIO_PULL_UP)>;

>         steps = <20>;

Reviewer: ```suggestion
        steps = <80>;
```
How about changing the example here to a common default?

Reviewer: Also, can you fix the indentation so the first line is consistent with the other lines?

        status = "disabled";
    };
```

> Here you will have to replace PIN_A and PIN_B with the appropriate pins that your PCB utilizes for the encoder(s). For keyboards that use the Pro Micro or any of the Pro Micro replacements, Sparkfun's [Pro Micro Hookup Guide](https://learn.sparkfun.com/tutorials/pro-micro--fio-v3-hookup-guide/hardware-overview-pro-micro) has a pinout diagram that can be useful to determine the right pins. Reference either the blue numbers labeled "Arduino" (digital pins) or the green numbers labeled "Analog" (analog pins). For pins that are labeled as both digital and analog, refer to your specific board's .dtsi file to determine how you should refer to that pin.

Reviewer: ```suggestion
Here you need to replace `PIN_A` and `PIN_B` with the appropriate pins that your PCB utilizes for the encoder(s). See [shield overlays section above](#shield-overlays) on the appropriate node label and pin number to use for GPIOs.
```
This seems a bit outdated, what do you think of this? (Also need to double check the section link works.)

Author: I like it. Merged this in, and confirmed in the deploy previous that the section link does work as expected.

> The `steps` property should corresponded to the documented pulses per rotation for the encoders used on the keyboard. If users use different encoders when they build, the value can be overridden in their keymap.

Reviewer: ```suggestion
The `steps` property should corresponded to the documented pulses per rotation for the encoders used on the keyboard, typically found on the datasheet of the component. If users use different encoders when they build, the value can be overridden in their keymap.
```
Would it be fair and/or useful to point this out specifically?
